### PR TITLE
Feature/epo 58/141

### PR DESCRIPTION
**Pull request by Dragos0000**

### CODE REVIEW

File: config/config-parameters.xsl [hunk 1]
```diff
@@ -21,20 +21,20 @@
     </xd:doc>
```
Comment (costezki) [inline config/config-parameters.xsl]: I would add an additional subfolder in the /config folder, indicating that this config is specifically destined to procurement project

File: config/config-parameters.xsl [hunk 2]
```diff
@@ -64,4 +64,16 @@
     <xsl:variable name="stereotypeValidOnEnumerations" select="()"/>
     <xsl:variable name="stereotypeValidOnPackages" select="()"/>
 
+    <!--Allowed characters for a normalized string-->
+    <xsl:variable name="allowedStrings" select="'^[\w\d-_:]+$'"/>
```
Comment (costezki) [inline config/config-parameters.xsl]: This variable shall disappear after the functionality refactoring. Add a refactoring ticket, please.

File: config/config-parameters.xsl [hunk 3]
```diff
@@ -64,4 +64,16 @@
...
+
+    <!-- Uppercased letters -->
+    <xsl:variable name="uppercaseLetters" select="'ABCDEFGHIJKLMNOPQRSTUVWXYZ'"/>
```
Comment (costezki) [inline config/config-parameters.xsl]: idem.

File: config/config-parameters.xsl [hunk 4]
```diff
@@ -64,4 +64,16 @@
...
+    <!--    Module URI for datashape definitions for the eProcurement ontology-->
+    <xsl:variable name="moduleUriShapes" select="fn:concat($base-uri, '/', 'ds')"/>
```
Comment (costezki) [inline config/config-parameters.xsl]: it is shapeModuleURI, RestrictionsModuleURI, coreModuleURI

File: src/common/fetchers.xsl [hunk 5]
```diff
@@ -20,7 +20,7 @@
         </xd:desc>
     </xd:doc>
     
-    <xsl:import href="../config-parameters.xsl"/>
+    <xsl:import href="../../config/config-parameters.xsl"/>
```
Comment (costezki) [inline src/common/fetchers.xsl]: how can we decouple the config file from the sources? 
We shall discuss that. The current approach is not scalable.

File: src/common/utils.xsl [hunk 6]
```diff
@@ -21,7 +21,7 @@
         </xd:desc>
     </xd:doc>
 
-    <xsl:import href="../config-parameters.xsl"/>
+    <xsl:import href="../../config/config-parameters.xsl"/>
```
Comment (costezki) [inline src/common/utils.xsl]: Idem ...

File: test/features/naming_policies.xspec [hunk 7]
```diff
@@ -1,8 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<x:description xmlns:x="http://www.jenitennison.com/xslt/xspec" schematron="file:model2owl/test/features/">
```
Comment (costezki) [inline test/features/naming_policies.xspec:2]: we have no acceptance tests in the /features folder? then probably can be deleted.

File: test/unitTests/test-common/test-utils.xspec [hunk 8]
```diff
@@ -106,47 +104,47 @@
     <x:scenario label="Scenario 1 for testing function getXsdRdfDataTypeValues">
         <x:call function="f:getXsdRdfDataTypeValues">
             <x:param name="qname">xsd:negativeInteger</x:param>
-            <x:param name="dataTypesDefinitions" href="../../testData/xsdAndRdfDataTypes.xml" select="/"/>
+            <x:param name="dataTypesDefinitions" href="../../../config/xsdAndRdfDataTypes.xml" select="/"/>
```
Comment (costezki) [inline test/unitTests/test-common/test-utils.xspec]: Idem, some flexible config loading mechanism needed, that decouples the code and tests from the final config file

File: test/unitTests/test-common/test-utils.xspec [hunk 9]
```diff
@@ -106,47 +104,47 @@
...
         </x:call>
         <x:expect label="data-type" select="'xsd:negativeInteger'"/>
     </x:scenario>
 
     <x:scenario label="Scenario 2 for testing function getXsdRdfDataTypeValues">
         <x:call function="f:getXsdRdfDataTypeValues">
             <x:param name="qname">xsd:!#4354gYearMonth</x:param>
-            <x:param name="dataTypesDefinitions" href="../../testData/xsdAndRdfDataTypes.xml" select="/"/>
+            <x:param name="dataTypesDefinitions" href="../../../config/xsdAndRdfDataTypes.xml" select="/"/>
```
Comment (costezki) [inline test/unitTests/test-common/test-utils.xspec]: this config, should probably a test configuration, to make sure it is not changing and that out tests still pass.

**Review by costezki — CHANGES_REQUESTED** [review]
well done! 

Some changes you can do already. The other ones about the decoupling of code and config, we still have to discuss.

File: config/config-parameters.xsl (hunk 2 above)
Comment (Dragos0000) [reply config/config-parameters.xsl]: In my opinion this should be left in here as they might change their mind to accept a new symbol in the name or to take out for example the digits.

File: config/config-parameters.xsl (hunk 1 above)
Comment (Dragos0000) [reply config/config-parameters.xsl]: Why as all files in that folder is config for the project.

File: config/config-parameters.xsl (hunk 3 above)
Comment (Dragos0000) [reply config/config-parameters.xsl]: created a ticket

File: test/features/naming_policies.xspec (hunk 7 above)
Comment (Dragos0000) [reply test/features/naming_policies.xspec:2]: this already deleted

File: test/unitTests/test-common/test-utils.xspec (hunk 8 above)
Comment (Dragos0000) [reply test/unitTests/test-common/test-utils.xspec]: done

File: test/unitTests/test-common/test-utils.xspec (hunk 9 above)
Comment (Dragos0000) [reply test/unitTests/test-common/test-utils.xspec]: done

File: config/config-parameters.xsl (hunk 1 above)
Comment (costezki) [reply config/config-parameters.xsl]: well, we might have other projects in the future, so we might have to think about how to extend and modularise checked and generators.

File: config/config-parameters.xsl (hunk 2 above)
Comment (costezki) [reply config/config-parameters.xsl]: oh! true, my mistake, this parameter has to be, then documented, what it does.

File: config/config-parameters.xsl (hunk 1 above)
Comment (costezki) [reply config/config-parameters.xsl]: We will have other projects in the future

File: src/common/checkers.xsl [hunk 10]
```diff
@@ -18,9 +18,8 @@
     </xd:doc>
 
     <xsl:import href="utils.xsl"/>
-
-    <xsl:variable name="allowedCharacters" select="'[a-zA-Z0-9-_:]'"/>
-    <xsl:variable name="allowedStrings" select="'^[\w\d-_:]+$'"/>
+    <xsl:import href="../../config/config-parameters.xsl"/>
+    
     <xsl:variable name="uppercaseLetters" select="'ABCDEFGHIJKLMNOPQRSTUVWXYZ'"/>
```
Comment (costezki) [inline src/common/checkers.xsl:23]: This should disappear, and instead, use a function "isUpperCase"

**Review by costezki — APPROVED** [review]
well done

**Pull request merged**